pkg/resource/deploy/step_generator.go

```diff
@@ -44,6 +44,9 @@ import (
 type stepGenerator struct {
 	deployment *Deployment // the deployment to which this step generator belongs
 
+	// true if this is a destroy generation
+	isDestroy bool
```

Comment:
Wondering if we should pre-emptively make this an enum (update/destroy, with refresh coming later) where the default is update?

Reply:
Yeh I did think that, was a choice of doing it now or doing the small fixup to when adding refresh later.